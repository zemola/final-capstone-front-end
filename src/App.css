* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.selected {
  background-color: black;
}

.link {
  font-size: 10px;
  padding: 5px;
  background-color: #eee;
  border: solid 1px #999;
  border-radius: 50px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-float infinite 3s ease-in-out;
  }
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: rgb(112, 76, 182);
}

@keyframes App-logo-float {
  0% {
    transform: translateY(0);
  }

  50% {
    transform: translateY(10px);
  }

  100% {
    transform: translateY(0);
  }
}
