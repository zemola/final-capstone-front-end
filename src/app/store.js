import { configureStore } from '@reduxjs/toolkit';
import carReducer from '../features/car/carSlice';
import userReducer from '../features/user/userSlice';
import reservationSlice from '../features/reservations/reservationSlice';
import singleCarSlice from '../features/singleCar/singleCarSlice';

const store = configureStore({
  reducer: {
    users: userReducer,
    cars: carReducer,
    reservations: reservationSlice,
    singleCar: singleCarSlice,
  },
});

export default store;
