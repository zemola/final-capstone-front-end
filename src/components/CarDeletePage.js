import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { deleteCar, fetchCars } from '../features/car/carSlice';

import styles from './CarDeletePage.module.css';

export default function CarDeletePage() {
  const dispatch = useDispatch();
  const { allCars } = useSelector((state) => state.cars);
  React.useEffect(() => {
    dispatch(fetchCars());
  }, [dispatch, allCars]);

  let list = null;
  if (allCars.length > 0) {
    list = allCars.map((car) => (
      <tr key={car.id} className={styles.tableRow}>
        <td className={styles.tableData}>
          {car.brand}
        </td>
        <td>
          <button type="button" className={styles.deleteBtn} aria-label="delete button" onClick={() => dispatch(deleteCar(car.id))}>
            Delete
          </button>
        </td>
      </tr>
    ));
  }

  return (
    <div className={styles.container}>
      <h1>Which car do you want to delete?</h1>
      <table className={styles.table}>
        <thead className={styles.tableHeading}>
          <tr className={styles.tableHeader}>
            <th>Car Name</th>
            <th>Delete</th>
          </tr>
        </thead>
        <tbody className={styles.tableBody}>
          {list}
        </tbody>
      </table>
    </div>
  );
}
