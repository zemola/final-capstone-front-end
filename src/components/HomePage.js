import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import CarCard from './CarCard';
import styles from './HomePage.module.css';

const HomePage = () => {
  const { cars } = useSelector((state) => state.cars);

  const showCars = () => {
    let carsList;
    if (cars.length > 0) {
      carsList = cars.map((item) => (
        <Link className={styles.carLink} state={item.id} to={`cars/${item.id}`} key={item.id}>
          <CarCard
            price={item.price}
            brand={item.brand}
            model={item.model}
          />
        </Link>
      ));
    } else {
      carsList = <h1 className={styles.noCars}>There are no cars yet</h1>;
    }
    return carsList;
  };

  useEffect(() => {
    showCars();
  });

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <h1 className={styles.mainHeading}>
          latest models
        </h1>
        <p className={styles.headingInfo}>
          Please select a model
        </p>
      </div>

      <div className={styles.modelsContainer}>
        <button type="button" onClick={() => console.log('Prev')} className={styles.pageBtnPrev}>
          <p>
            {'<'}
          </p>
        </button>
        <div className={styles.carsContainer}>
          { showCars() }
        </div>
        <button type="button" onClick={() => console.log('Next')} className={styles.pageBtnNext}>
          <p>
            {'>'}
          </p>
        </button>
      </div>
    </div>
  );
};

export default HomePage;
