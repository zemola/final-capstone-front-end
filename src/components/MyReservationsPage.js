import React from 'react';
import { useSelector } from 'react-redux';

import styles from './MyReservationsPage.module.css';

export default function MyReservationsPage() {
  const { reservations } = useSelector((state) => state.reservations);

  let list = null;
  if (reservations.length > 0) {
    list = reservations.map((resv) => (
      <tr key={resv.id} className={styles.tableRow}>
        <td className={styles.tableData}>
          {resv.car.brand}
        </td>
        <td className={styles.tableData}>
          {resv.car.reservation_from}
        </td>
        <td className={styles.tableData}>
          {resv.car.reservation_until}
        </td>
      </tr>
    ));
  }

  return (
    <div className={styles.container}>
      <h1>Which car do you want to delete?</h1>
      <table className={styles.table}>
        <th className={styles.tableHeading}>
          <td>Car Name</td>
          <td>Reserved From</td>
          <td>Reserved Until</td>
          <td>Delete</td>
        </th>
        {list}
      </table>
    </div>
  );
}
