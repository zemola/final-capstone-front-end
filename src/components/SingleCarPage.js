import React, { useEffect } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import styles from './SingleCarPage.module.css';
import carImage from '../img/car.png';
import { fetchSingleCar } from '../features/singleCar/singleCarSlice';

export default function SingleCarPage() {
  const location = useLocation();
  const id = location.state;
  const dispatch = useDispatch();
  const { car } = useSelector((state) => state.singleCar);
  useEffect(() => {
    dispatch(fetchSingleCar(id));
  }, [dispatch, id]);

  return (
    <div className={styles.container}>
      <div className={styles.carImageContainer}>
        <img className={styles.carImage} src={carImage} alt="carImage" />
        <Link to="/" className={styles.pageBtnPrev}>
          <p>{'<'}</p>
        </Link>
      </div>
      <div className={styles.carDetails}>
        <div className={styles.carNameContainer}>
          <h3>{car.brand}</h3>
          <p>{car.model}</p>
        </div>
        <ul className={styles.infoChart}>
          <li
            style={{ backgroundColor: '#e2e3e5' }}
            className={styles.infoChartItem}
          >
            <p className={styles.infoChartText}>Price</p>
            <p className={styles.infoChartText}>
              $
              {car.price}
            </p>
          </li>
          <li className={styles.infoChartItem}>
            <p className={styles.infoChartText}>Year</p>
            <p className={styles.infoChartText}>{car.release_year}</p>
          </li>
          <li
            style={{ backgroundColor: '#e2e3e5' }}
            className={styles.infoChartItem}
          >
            <p className={styles.infoChartText}>Seats</p>
            <p className={styles.infoChartText}>{car.seats}</p>
          </li>
          <li className={styles.infoChartItem}>
            <p className={styles.infoChartText}>Wheels</p>
            <p className={styles.infoChartText}>{car.wheel_drive}</p>
          </li>
        </ul>
        <div className={styles.btnContainer}>
          <Link to="/reserve" className={styles.btn} state={car.id}>
            Reserve
          </Link>
        </div>
      </div>
    </div>
  );
}
