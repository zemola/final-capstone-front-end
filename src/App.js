import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Outlet } from 'react-router-dom';
import './App.css';
import { fetchUser } from './features/user/userSlice';
import { fetchReservatins } from './features/reservations/reservationSlice';
import SideBar from './components/SideBar';
import LoginPage from './components/LoginPage';

function App() {
  const dispatch = useDispatch();
  const { currentUser } = useSelector((state) => state.users);

  const isUser = () => {
    if (!currentUser) {
      return (
        <LoginPage />
      );
    }
    return (
      <>
        <SideBar />
        <Outlet />
      </>
    );
  };

  React.useEffect(() => {
    dispatch(fetchUser());
    dispatch(fetchReservatins());
  }, [dispatch]);

  return (
    <div className="App">
      {isUser()}
    </div>
  );
}

export default App;
