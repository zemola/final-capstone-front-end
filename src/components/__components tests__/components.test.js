import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../app/store';
import App from '../../App';
import HomePage from '../HomePage';
import SingleCarPage from '../SingleCarPage';
import MyReservationsPage from '../MyReservationsPage';
import CarDeletePage from '../CarDeletePage';

describe('Test all app Components', () => {
  it('Test the app landing Page', () => {
    const home = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route element={<App />}>
                <Route element={<HomePage />} index />
              </Route>
            </Routes>
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(home).toMatchSnapshot();
  });

  it('Test the app Single car Page', () => {
    const carDetails = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route element={<App />}>
                <Route element={<SingleCarPage />} />
              </Route>
            </Routes>
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(carDetails).toMatchSnapshot();
  });

  it('Test the app Reserve car Page', () => {
    const reserve = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route element={<App />}>
                <Route element={<MyReservationsPage />} />
              </Route>
            </Routes>
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(reserve).toMatchSnapshot();
  });

  it('Test the app Delete car Page', () => {
    const Delete = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route element={<App />}>
                <Route element={<CarDeletePage />} />
              </Route>
            </Routes>
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(Delete).toMatchSnapshot();
  });
});
