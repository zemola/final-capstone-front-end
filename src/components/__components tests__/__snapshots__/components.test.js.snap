// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all app Components Test the app Delete car Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test all app Components Test the app Reserve car Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test all app Components Test the app Single car Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test all app Components Test the app landing Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="container"
        >
          <img
            alt="app logo"
            class="logoImg"
            src="logo.png"
          />
          <h1
            class="heading"
          >
            Wellcome to Our Page
          </h1>
          <div
            class="inputesContainer"
          >
            <input
              class="input"
              placeholder="Enter your username"
              required=""
              type="text"
              value=""
            />
            <input
              class="input"
              placeholder="Enter your email"
              required=""
              type="email"
              value=""
            />
            <button
              class="button"
              type="button"
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="container"
      >
        <img
          alt="app logo"
          class="logoImg"
          src="logo.png"
        />
        <h1
          class="heading"
        >
          Wellcome to Our Page
        </h1>
        <div
          class="inputesContainer"
        >
          <input
            class="input"
            placeholder="Enter your username"
            required=""
            type="text"
            value=""
          />
          <input
            class="input"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
          <button
            class="button"
            type="button"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
